src/physics/physics_model.cpp

```diff
@@ -159,12 +225,25 @@ real PhysicsModel<dim,nstate,real,nstate_baseline_physics>
 ::max_convective_eigenvalue (const std::array<real,nstate> &conservative_soln) const
 {
     // TO DO: Update for when nstate > nstate_baseline_physics
+    //real max_eig;
+    //if constexpr(nstate==nstate_baseline_physics) {
+    //    max_eig = physics_baseline->max_convective_eigenvalue(conservative_soln);
+    //} else {
+    //    // TO DO, make use of the physics_model object for nstate>nstate_baseline_physics
+    //    std::abort();
+    //}
+
     real max_eig;
-    if(nstate==nstate_baseline_physics) {
+    if constexpr(nstate==nstate_baseline_physics) {
         max_eig = physics_baseline->max_convective_eigenvalue(conservative_soln);
     } else {
-        // TO DO, make use of the physics_model object for nstate>nstate_baseline_physics
-        std::abort();
+        max_eig = model->max_convective_eigenvalue(conservative_soln);
+        std::array<real,nstate_baseline_physics> baseline_conservative_soln;
+        for(int s=0; s<nstate_baseline_physics; ++s){
+            baseline_conservative_soln[s] = conservative_soln[s];
+        }
+        real baseline_max_eig = physics_baseline->max_convective_eigenvalue(baseline_conservative_soln);
+        max_eig = max_eig > baseline_max_eig ? max_eig : baseline_max_eig;
```

Comment:
So this means that the model's max convective eigenvalue represents the total convective_flux (baseline's + model's convective flux) ?

Reply:
I take the maximum max_eig between the one from the baseline and the model.

Reply:
Any reason for the ternary operator rather than std::max? Is it due to AD?

Reply:
@dougshidong in fact there is no special reason, do you suggest using std::max?

Reply:
Will have to modify this according to comment https://github.com/dougshidong/PHiLiP/pull/172/files#r988099447

Reply:
As discussed with Siva, this part works fine in the case of zero additional convective flux in baseline physics, so I only leave a note mentioning that.